services/QuillLMS/spec/models/teacher_activity_feed_spec.rb

```diff
@@ -0,0 +1,166 @@
+
+        expect(text).to eq(ActivitySession::PROFICIENT)
+      end
+    end
+
+    context 'when the activity session percentage is at or above the nearly proficient cutoff' do
+      it "returns #{ActivitySession::NEARLY_PROFICIENT}" do
+        text = feed.send(:text_for_score, nearly_proficient_activity_session.classification.key, nearly_proficient_activity_session.percentage)
+
+        expect(text).to eq(ActivitySession::NEARLY_PROFICIENT)
+      end
+    end
+
+    context 'when the activity session percentage is below the nearly proficient cutoff' do
+      it "returns #{ActivitySession::NOT_YET_PROFICIENT}" do
+        text = feed.send(:text_for_score, not_yet_proficient_activity_session.classification.key, not_yet_proficient_activity_session.percentage)
+
+        expect(text).to eq(ActivitySession::NOT_YET_PROFICIENT)
+      end
+    end
+
+  end
+
+  describe '#text_for_completed' do
+    let(:feed) { TeacherActivityFeed.new('fake_id') }
+
+    context 'when the timestamp was less than one minute ago' do
+      it 'should return "0 mins ago"' do
+        expect(feed.send(:text_for_completed, 30.seconds.ago)).to eq("0 mins ago")
+      end
+    end
+
+    context 'when the timestamp was one minute ago' do
+      it 'should return "1 min ago"' do
+        expect(feed.send(:text_for_completed, 1.minute.ago)).to eq("1 min ago")
+      end
+    end
+
+    context 'when the timestamp was more than one minute but less than one hour ago' do
+      it 'should return "47 min ago"' do
+        expect(feed.send(:text_for_completed, 47.minutes.ago)).to eq("47 mins ago")
+      end
+    end
+
+    context 'when the timestamp was one hour ago' do
+      it 'should return "1 hour ago"' do
+        expect(feed.send(:text_for_completed, 1.hour.ago)).to eq("1 hour ago")
+      end
+    end
+
+    context 'when the timestamp was more than one hour but less than a day ago' do
+      it 'should return "22 hours ago"' do
+        expect(feed.send(:text_for_completed, 22.hours.ago)).to eq("22 hours ago")
+      end
+    end
+
+    context 'when the timestamp was one day ago' do
+      it 'should return "1 day ago"' do
+        expect(feed.send(:text_for_completed, 1.day.ago)).to eq("1 day ago")
+      end
+    end
+
+    context 'when the timestamp was more than one day but less than one week ago' do
+      it 'should return "3 days ago"' do
+        expect(feed.send(:text_for_completed, 3.days.ago)).to eq("3 days ago")
+      end
+    end
+
+    context 'when the timestamp was more than one week ago but this calendar year' do
+      it 'should return the written out date' do
+        date = 8.days.ago
+        if date.year == Time.now.year
+          expect(feed.send(:text_for_completed, date)).to eq(date.strftime("%b #{date.day.ordinalize}"))
+        else
+          expect(feed.text_for_completed(date)).to eq(date.strftime("%b #{date.day.ordinalize}, %Y"))
```

Comment:
I think this branch will fail without the same `.send` pattern used elsewhere in this spec due to the fact that `text_for_completed` is private.

Reply:
Yeah, breaking out this branch into its own test and converting this to a pure function for easier testing.